packages/react/src/ReactSharedInternals.js

```diff
@@ -15,6 +15,8 @@ import ReactDebugCurrentFrame from './ReactDebugCurrentFrame';
 const ReactSharedInternals = {
   ReactCurrentDispatcher,
   ReactCurrentOwner,
+  // used by act()
+  isActingUpdates: [false],
```

Comment:
Let's use the same ref-style pattern as ReactCurrentOwner and ReactCurrentDispatcher:

```js
// Pick off the shared internals object
const {ReactShouldWarnActingUpdates} = ReactSharedInternals;

// Mutate like a ref
ReactShouldWarnActingUpdates.current = true;
```